### PR TITLE
Feature/wfone 2656

### PR DESCRIPTION
**Pull request by vivid-fxie**
Update PublicPublishedIncidentEndpoint to filter result by discoveryDate

### CODE REVIEW

File: server/wfnews-persistence/src/main/resources/ca/bc/gov/nrs/wfnews/persistence/v1/dao/mybatis/mapper/PublishedIncidentMapper.xml [hunk 1]
```diff
@@ -333,6 +333,12 @@
 			<if test="fireCentreName != null &amp;&amp; fireCentreName != ''">  
 				AND fire_centre_name = #{fireCentreName, javaType=java.lang.String, jdbcType=VARCHAR, mode=IN}
 			</if>
+			<if test="discoveryDate != null">
+				AND DISCOVERY_DATE > #{discoveryDate, javaType=java.util.Date, jdbcType=DATE, mode=IN}
```
Comment (dhlevi) [inline server/wfnews-persistence/src/main/resources/ca/bc/gov/nrs/wfnews/persistence/v1/dao/mybatis/mapper/PublishedIncidentMapper.xml:337]: You dont want these to be `>=` ?

File: server/wfnews-api-rest-endpoints/src/main/java/ca/bc/gov/nrs/wfnews/api/rest/v1/endpoints/PublicPublishedIncidentEndpoint.java [hunk 2]
```diff
@@ -37,6 +37,8 @@ public Response getPublishedIncidentList(
 		@ApiParam("Filter on fires that are new within 24 hours") @QueryParam("newFires") Boolean newFires,
 		@ApiParam("Filter on fire centre code") @QueryParam("fireCentreCode") String fireCentreCode,
 		@ApiParam("Filter on fire centre name") @QueryParam("fireCentreName") String fireCentreName,
+		@ApiParam("Filter on discovery date") @QueryParam("discoveryDate") String discoveryDate,
```
Comment (dhlevi) [inline server/wfnews-api-rest-endpoints/src/main/java/ca/bc/gov/nrs/wfnews/api/rest/v1/endpoints/PublicPublishedIncidentEndpoint.java:40]: The query params "discoveryDate" and "createDate" do not make it clear that its actually doing a "greater than check". 

- Please rename the attributes so they identify what the param is actually for, or...
-  optionally provide "from/to" dates for discovery and create date... or...
- Make it an equals, not a greater than

**Review by dhlevi — CHANGES_REQUESTED** [review]

File: server/wfnews-api-rest-endpoints/src/main/java/ca/bc/gov/nrs/wfnews/api/rest/v1/endpoints/PublicPublishedIncidentEndpoint.java (hunk 2 above)
Comment (vivid-fxie) [reply server/wfnews-api-rest-endpoints/src/main/java/ca/bc/gov/nrs/wfnews/api/rest/v1/endpoints/PublicPublishedIncidentEndpoint.java:40]: I will address this

File: server/wfnews-persistence/src/main/resources/ca/bc/gov/nrs/wfnews/persistence/v1/dao/mybatis/mapper/PublishedIncidentMapper.xml (hunk 1 above)
Comment (vivid-fxie) [reply server/wfnews-persistence/src/main/resources/ca/bc/gov/nrs/wfnews/persistence/v1/dao/mybatis/mapper/PublishedIncidentMapper.xml:337]: Active fire monitor insert the newest fire's ignition date(will be changed to insert createDate) as last_fetched_time_stamp, add = may get fires that were already reported during last fetch period.

**Comment by dhlevi** [comment]
@vivid-fxie Changes are good, but I just want to make really sure about this change, as the inclusion of only "greater than" without something to apply for an upper bounds seems strange.

Should this be Greater Than and not >=? if so, why?
Do we not need an upper bounds (ie, between dates x and y)?

Don't think about the change from the perspective of what PM needs, but what's best for the API itself.

**Review by dhlevi — APPROVED** [review]

**Pull request merged**